x-pack/plugins/reporting/server/lib/content_stream.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Duplex } from 'stream';
+import type { ElasticsearchClient } from 'src/core/server';
+import { ReportDocument } from '../../common/types';
+
+type Callback = (error?: Error) => void;
+type SearchRequest = Required<Parameters<ElasticsearchClient['search']>>[0];
+
+export interface ContentStreamDocument {
+  id: string;
+  index: string;
+  if_primary_term?: number;
+  if_seq_no?: number;
+}
+
+export class ContentStream extends Duplex {
+  private buffer = '';
+  private primaryTerm?: number;
+  private seqNo?: number;
+
+  constructor(private client: ElasticsearchClient, private document: ContentStreamDocument) {
+    super();
+  }
+
+  async _read() {
+    const { id, index } = this.document;
+    const body: SearchRequest['body'] = {
+      _source: { includes: ['output.content'] },
+      query: {
+        constant_score: {
+          filter: {
+            bool: {
+              must: [{ term: { _id: id } }],
```

Comment:
We need to add the filter that was in `jobsQueries.getContent`, which was to match the `user` of the document with the authenticated user. 

This would preserve the requirement that users can not download reports created by other users.

Reply:
I added that at first, but then I decided not to do that because I don't think it belongs there. The stream itself should be as simple as possible and responsible only for reading and writing the data. And apart from that, we already have this check in the store and perform that [here](https://github.com/elastic/kibana/pull/106821/files#diff-aefadc0f450e55433b76a1bcbb14f7d7720535eb78c52720b507f53f385b7212R37) before reading from the stream.

Reply:
++ after I wrote my previous comment, I realized that `.get` is checking the username for us.